### PR TITLE
CIEDEV-495 Add Tag Manager - inital commit

### PR DESCRIPTION
**Pull request by niknik0**

### CODE REVIEW

File: internal/module/services/registrar.go [hunk 1]
```diff
@@ -30,7 +30,7 @@ var (
 	RegistrarTableName       string = DefaultRegistrarTableName
 	RegistrarServiceEndpoint string = DefaultRegistrarServiceEndpoint
 
-	ModuleRegistered = &terrarium.Response{Message: "Module registered successfully."}
+	ModuleTagPublished = &terrarium.Response{Message: "Module registered successfully."}
```
Comment (odelanno) [inline internal/module/services/registrar.go:33]: That should not have been renamed.

File: internal/module/services/registrar.go [hunk 2]
```diff
@@ -139,7 +139,7 @@ func (s *RegistrarService) Register(ctx context.Context, request *terrarium.Regi
 	}
 
 	log.Println("New module registered.")
-	return ModuleRegistered, nil
+	return ModuleTagPublished, nil
```
Comment (odelanno) [inline internal/module/services/registrar.go:142]: as above

**Review by odelanno — DISMISSED** [review]

**Review by odelanno — DISMISSED** [review]

File: internal/module/services/tag_manager.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		log.Println(err)
+		return ModuleTagTableInitializationError
+	}
+
+	RegisterTagManagerServer(grpcServer, s)
+
+	return nil
+}
+
+func (s *RegistrarService) PublishTag(ctx context.Context, request *terrarium.PublishTagRequest) (*terrarium.Response, error) {
```
Comment (vmilovanovicc) [inline internal/module/services/tag_manager.go:66]: Shouldn't this refer to TagManagerService like `s *TagManagerService`?

File: internal/module/services/tag_manager.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		av, err := dynamodbattribute.MarshalMap(ms)
+
+		if err != nil {
+			log.Println(err)
+			return nil, MarshalModuleError
+		}
+
+		in := &dynamodb.PutItemInput{
+			Item:      av,
+			TableName: aws.String(RegistrarTableName),
```
Comment (vmilovanovicc) [inline internal/module/services/tag_manager.go:98]: We should refer to the tags table as we're storing module tags in the tags table.  
```
TableName: aws.String(TagTableName)
```

File: internal/module/services/tag_manager.go [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+		update.Set(expression.Name("modified_on"), expression.Value(time.Now().UTC().String()))
+		expr, err := expression.NewBuilder().WithUpdate(update).Build()
+
+		if err != nil {
+			log.Println(err)
+			return nil, ExpressionBuildError
+		}
+
+		in := &dynamodb.UpdateItemInput{
+			TableName: aws.String(RegistrarTableName),
```
Comment (vmilovanovicc) [inline internal/module/services/tag_manager.go:116]: When we update the tags, the records in the tags table should be updated.

```
TableName: aws.String(TagTableName)
```

**Review by odelanno — DISMISSED** [review]

**Review by odelanno — APPROVED** [review]

**Pull request merged**